benchmark/Dockerfile

```diff
@@ -6,18 +9,12 @@ RUN hail-apt-get-install \
     openjdk-11-jre-headless \
     wget
 
-COPY benchmark/python/pinned-requirements.txt benchmark-requirements.txt
-COPY hail/python/pinned-requirements.txt hail-requirements.txt
-RUN hail-pip-install \
-    -r benchmark-requirements.txt \
-    -r hail-requirements.txt
-
-ARG BENCHMARK_WHEEL
-COPY ${BENCHMARK_WHEEL} .
-
-ARG HAIL_WHEEL
-COPY ${HAIL_WHEEL} .
+RUN --mount=src=benchmark/python/pinned-requirements.txt,target=benchmark-requirements.txt \
+    --mount=src=hail/python/pinned-requirements.txt,target=hail-requirements.txt \
+    hail-pip-install -r benchmark-requirements.txt -r hail-requirements.txt
 
-RUN hail-pip-install --no-deps --quiet \
-    $(basename ${BENCHMARK_WHEEL}) \
-    $(basename ${HAIL_WHEEL})
+ARG HAIL_WHEEL_DIR
+RUN --mount=src=${HAIL_WHEEL_DIR},target=${HAIL_WHEEL_DIR} \
+    --mount=src=benchmark/python,target=/benchmark,rw \
```

Comment:
I saw errors about read-only filesystems when I did this. Do you think it's OK to mount instead of copy? I'm not sure what files get written and if they're compatible across OSs....

Reply:
I think this is safe. As the [documentation says](https://docs.docker.com/engine/reference/builder/#run---mounttypebind) and [this issue](https://github.com/moby/buildkit/issues/903) elaborates on, any files written in the `rw` mount are not really persisted (they're written to a temporary writable layer on top of the docker context), and seems to be for situations like these where your `RUN` needs some scratch space.